hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistProxyFactory.java

```diff
@@ -93,7 +93,21 @@ public void postInstantiate(
 		javassist.util.proxy.ProxyFactory factory = new javassist.util.proxy.ProxyFactory() {
 			@Override
 			protected ClassLoader getClassLoader() {
-				return persistentClass.getClassLoader();
+				// This classloader does not take into account class space consistency
+				// i.e. if the persistenceClass can see the same interface types via
+				// a different ClassLoader hierarchy. In that case we could try some 
+				// risky magic with loadClass, but using findClass is much safer.
+				return new ClassLoader(persistentClass.getClassLoader()) {
```

Comment:
@sebersole, what's your opinion of this approach?  Technically, we could work in access to the `ClassLoaderService` here, since it's available when `postInstantiate` is called (the Tuplizers have the `EntityMetamodel`).  Although, admittedly, I'm a little shaky on the current state of our classloading.

@timothyjward, I *think* that would accomplish essentially the same thing, but would be more consistent with our classloading approach.

Correct me if I'm wrong in any of that.  Either way, great suggestion -- getting rid of the requirement to import those packages within the client bundle has been a priority...